_posts/2020-09-15-open-sourcing-a-private-project.markdown

```diff
@@ -0,0 +1,89 @@
+
+Follow steps on [GitHub](https://docs.github.com/en/github/getting-started-with-github/create-a-repo) to create a new repository and make sure to choose the public option when choosing the visibility, and don’t include the README.
+```
+# Choose which commit you would like to include in the new tree(we chose the first commit). Use git log and copy the commit hash.
+$ git reset --soft <commit hash>
+$ git commit --amend
+# adjust commit title as needed
+$ git remote rename origin <oldrepo>
+$ git remote add origin <clone new GitHub repository>
+$ git push origin <main branch>
+```
+If you are looking to squash the last number of commits into a single one in the commit history, we can use the git reset --soft [commit] option. In our case we squashed all commits against the first one, but deleting the repo and recreating it would have given the same effect. There's an interesting side effect in this approach though, GitHub seems to detect contributions from other developers while if it was a repo from scratch it wouldn't.
+
+## Creating a GitHub Page
+A GitHub page is not essential when open sourcing a project, but it is a good idea. This is a place where we can give more information to potential users, and even share screenshots of the project. Users will have an easier time setting up the application and understanding how it is used when they have a good resource to pull from. We have GitHub Pages for several of our projects including; [Dash](https://fastruby.github.io/dash/), [Points](https://fastruby.github.io/points/), and [Pecas](https://fastruby.github.io/pecas/).
+
+Steps we used to create a GitHub Page
+```
+$ git checkout --orphan gh-pages
+# preview files to be deleted
+$ git rm -rf --dry-run .
+# actually delete the files
+$ git rm -rf .
+# create an index.html file
+# add any styling you want to associate for the file
+# Push gh-pages branch to GitHub
+```
+Go into settings for the repository and activate a Gh-Page. It will automatically use your gh-pages branch.
+
+Note. This is for custom GitHub pages. It is also possible to quickly build a GitHub page using a theme and the already existing README.
+
+## Badges
+Adding Badges to the README is a great way to show pieces of information to potential users quickly and aesthetically. [Circle CI](https://circleci.com/docs/2.0/status-badges/), [Travis CI](https://docs.travis-ci.com/user/status-images/), [Code Climate](https://codeclimate.com/github/codeclimate/codeclimate/badges/) and more have badges or status images that you can embed in the README.
+
+## Adding known issues to GitHub
+
+We tend to use [Pivotal Tracker](https://www.pivotaltracker.com/) for our internal projects to keep track of stories and known issues. Once we open source we like to close the Pivotal Tracker project and move all known issues or desired features to GitHub Issues. This way it will be a public history of everything being worked on in the project.
```

Comment:
@FionaDL it would be nice to add a link to https://www.ombulabs.com/blog/agile/pivotal-tracker/how-we-use-pivotal-tracker-at-ombu-labs.html when you mention Pivotal Tracker